### PR TITLE
Use tomllib/tomli/tomli-w instead of unmaintained toml

### PR DESCRIPTION
**Pull request by mgorny**
Replace the unmaintained `toml` package with the modern replacements. Use the built-in `tomllib` module to load TOML files in Python 3.11, with fallback to `tomli` for older Python versions.  Use the `tomli-w` package to dump TOML files.  These packages feature correct TOML 1.0 support.

### CODE REVIEW

**Comment by cdgriffith** [comment]
Thanks for the contribution! Don't worry about the tests, have the fixes for those mypy updates in another PR

**Comment by mgorny** [comment]
> Thanks for the contribution! Don't worry about the tests, have the fixes for those mypy updates in another PR

Thanks. Please let me know if you want me to rebase the PR, so that they run cleanly (or just push to my branch).

File: box/converters.py [hunk 1]
```diff
@@ -195,24 +207,22 @@ def _from_yaml(
     return data
 
 
-def _to_toml(obj, filename: Union[str, PathLike] = None, encoding: str = "utf-8", errors: str = "strict"):
+def _to_toml(obj, filename: Union[str, PathLike] = None):
```
Comment (cdgriffith) [inline box/converters.py:210]: Can't change function signatures outside a major revision for Box, so will need to keep these as they were originally.

File: box/converters.py [hunk 2]
```diff
@@ -195,24 +207,22 @@ def _from_yaml(
...
+            tomli_w.dump(obj, f)
     else:
-        return toml.dumps(obj)
+        return tomli_w.dumps(obj)
 
 
-def _from_toml(
-    toml_string: str = None, filename: Union[str, PathLike] = None, encoding: str = "utf-8", errors: str = "strict"
-):
+def _from_toml(toml_string: str = None, filename: Union[str, PathLike] = None):
```
Comment (cdgriffith) [inline box/converters.py:219]: repeat of can't change function signatures

**Review by cdgriffith — COMMENTED** [review]
Thanks again for the contribution, I honestly didn't realize that `toml` has seem to have had it's maintainer offline for months. I see he's active again on twitter, so possibly will get back to it. That and the fact it's still much more popular than the other kits, I will still need to check and use that first if it's available on the system, then add tomli as the third fallback. 

Same as with the `toml` extra should remain as the `toml` library and a new `tomli` one should be added on its own. 

They may eventually switch to defaults with a major version release, but don't want to introduce a core library change at this point unless there is clear risk of using it.

File: box/converters.py (hunk 1 above)
Comment (mgorny) [reply box/converters.py:210]: I thought functions starting with underscore were supposed to be private.

**Comment by cdgriffith** [comment]
Going to merge this and add support for both as I have some time today

**Pull request merged**